### PR TITLE
Replace Buffer with Uint8Array in codec

### PR DESCRIPTION
**Pull request by Lbqds**
Currently, our codec uses `Buffer`, which is imported from `buffer/`. But it is not compatible with `Uint8Array`.

### CODE REVIEW

File: packages/web3/src/address/address.ts [hunk 1]
```diff
@@ -20,9 +20,9 @@ import { ec as EC } from 'elliptic'
...
 import blake from 'blakejs'
-import bs58 from './bs58'
-import djb2 from './djb2'
-import { binToHex, hexToBinUnsafe } from './utils'
+import bs58 from '../utils/bs58'
+import djb2 from '../utils/djb2'
+import { binToHex, hexToBinUnsafe } from '../utils'
 import { KeyType } from '../signer'
 import { MultiSig, lockupScriptCodec } from '../codec/lockup-script-codec'
 import { Buffer } from 'buffer/'
```
Comment (nop33) [inline packages/web3/src/address/address.ts:28]: Is this still needed?

File: packages/web3/src/codec/script-codec.test.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
+import { Buffer } from 'buffer/'
```
Comment (nop33) [inline packages/web3/src/codec/script-codec.test.ts:19]: and this?

File: packages/web3/src/codec/bytestring-codec.test.ts [hunk 3]
```diff
@@ -21,15 +21,15 @@ import { compactUnsignedIntCodec } from './compact-int-codec'
 
 describe('Encode & decode bytestring', function () {
   it('should encode & decode bytestring', function () {
-    success(Buffer.from(''))
-    success(Buffer.from('Alephium is great!'))
+    success(new Uint8Array([]))
+    success(new Uint8Array(Buffer.from('Alephium is great!')))
```
Comment (nop33) [inline packages/web3/src/codec/bytestring-codec.test.ts:25]: `Buffer` Could probably be removed from here as well?

**Review by nop33 — COMMENTED** [review]
Nice job man!

Just my 2 cents, once the need for `Buffer` is removed, how about cleaning up the webpack polyfills as well?

- https://github.com/alephium/alephium-web3/blob/replace-buffer-with-uint8array-in-codec/packages/web3/webpack.config.js#L28-L30
- https://github.com/alephium/alephium-web3/blob/replace-buffer-with-uint8array-in-codec/packages/web3/webpack.config.js#L44

**Comment by polarker** [comment]
@Lbqds Yes. Let's take the chance to remove `Buffer` completely from the project, not just `codec`.

**Comment by Lbqds** [comment]
Replaced all `Buffer` with `Uint8Array` in the web3 package.

**Review by polarker — APPROVED** [review]
LGTM!

**Pull request merged**